File: java/Reverse_Integer.java
// be careful with the overflow case 
// use the formula : result = result * 10 + bit ;
public class Solution {
    public int reverse(int x) {
        int result = 0, signal = 1;
        if (x == 0) return 0 ;
        if (x < 0) { 
            signal = -1 ;
            x = -x ;
        }
        while (x > 0) {
            int bit = x % 10 ;
            // deal with overflow
            if ((Integer.MAX_VALUE - bit) / 10 < result )
                return 0;
            result = result * 10 + bit ;
            x /= 10 ;
        }
        return result * signal ;
    }
}

File: Python/Reverse Integer.py
class Solution:
    # @param {integer} x
    # @return {integer}
    def reverse(self, x):
        sig = x<0 and -1 or 1
        x = abs(x)
        ret = 0
        while x:
            ret *= 10
            ret += x % 10
            x //= 10
        if ret > 2147483641 :
            ret = 0
        return ret * sig
